src/prefect/agent/kubernetes/agent.py

```diff
@@ -516,7 +516,9 @@ def generate_job_spec_from_environment(
         env[4]["value"] = self.namespace
         env[5]["value"] = str(self.labels)
         env[6]["value"] = str(self.log_to_cloud).lower()
-        env[7]["value"] = config.logging.level
+        env[7]["value"] = self.env_vars.get(
```

Comment:
Because we additionally populate all env vars in lines 524+ below, I'd vote to change this to 
```suggestion
        env[7]["value"] = self.env_vars.pop(
```

Reply:
I also think that we need to make similarly intentional changes to the other agents, e.g., here are some lines in the local agent that, if swapped, would break this feature: https://github.com/PrefectHQ/prefect/blob/master/src/prefect/agent/local/agent.py#L193-L214
and similarly for the docker and ecs agents.

Reply:
Hmmm yep I can get behind the idea of more explicit logic here, I'll make some updates

Reply:
@cicdw re:
> make similarly intentional changes to the other agents, e.g., here are some lines in the local agent that, if swapped, would break this feature

So not only would changing the order of lines break this feature, it would break explicitly established precedence patterns, which are well described in comments. This precedence is well tested for Docker and Local Agents.

Would adding tests for ECS and K8s agents + adding comments to `env.update(run_config.env)` calls be sufficient here?

It feels clunky to write something like
```python
env.update(run_config.env.pop('PREFECT__LOGGING__LEVEL'), env['PREFECT__LOGGING__LEVEL'])
env.update(run_config.env)
```

Reply:
Because logging level is a uniquely important / special environment variable that we plan to elevate as a dedicated option in the UI, we need to treat it as such in code as well so that future changes to how we handle the `.env` attribute on run configs doesn't result in hidden breakages of this feature.  So in this case, every agent should explicitly set this variable with some precedence rule as the K8s agent does (note that environment variable at position `7` is set explicitly as logging level in the yaml template).  

What that looks like will most likely be agent-dependent.